### PR TITLE
language: Add `string_delimiter_escape` to language definitions

### PR DESCRIPTION
**Pull request by pareksha**
This adds characters used to escape delimiters/quotes
for each language in their respective language
definitions.

This is part of `Convert Bear To Aspects` GSoC project.

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by pareksha** [comment]
@jayvdb what I want this field to have is the `escape_quotes` by which I literally mean escaping the quote. Best illustrated with the example where printing `"This is \"cool\"` in python and `"This is ""cool"""` in Fortran both output `This is "cool"`. This is how both fields relate. Do let me now if you agree with me so that I can proceed or the design needs to change.

One more question, where to add the documentation for `escape_quotes` (Also let me know whether the name feels right to you)?

**Comment by jayvdb** [comment]
Documentation should be done according to pep 257

**Comment by jayvdb** [comment]
Blocked on a decent language definition for at least Tcl, possibly m4 and VB https://github.com/coala/coala/issues/5578 , so that we can better see how escaping works.  Also we're finding lots of implicit limitations, which maybe can not be solved easily, but we can make them explicit limitations so we know we cant support certain languages due to properties of the language. (And this will also help the quickstart green project, as they can skip relevant bears for languages which it is unable to parse at all)

File: coalib/bearlib/languages/definitions/C.py [hunk 1]
```diff
@@ -19,3 +19,4 @@ class C:
         '#undef', '#ifdef', '#ifndef', '#if', '#endif', '#else', '#elif',
         '#line', '#pragma']
     special_chars = list(r'+-*/.;\,()[]{}\=<>|&^~?%!')
+    escape_quotes = '\"',
```
Comment (jayvdb) [inline coalib/bearlib/languages/definitions/C.py]: We can merge this if they all change to being dictionaries which describe how to escape string delimiters

`string_delimiter_escape = {'"': '\\"'}`

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5504/commits/abe011df134b9a1cec685f174deb6a223c8f1196

**Comment by jayvdb** [comment]
unack abe011d

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5504/commits/396fc76f92981e515c1e1771ec6b5ec341534c71

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5504/commits/ca677d3f4c81fb775a327858e172c21b55528dec

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**